lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/EvaluatorManager.cs

```diff
@@ -0,0 +1,311 @@
+        /// </summary>
+        /// <param name="evaluatorId"></param>
+        internal void RecordFailedEvaluator(string evaluatorId)
+        {
+            RomoveAllocatedEvaluator(evaluatorId);
+
+            if (_failedEvaluators.Contains(evaluatorId))
+            {
+                string msg = string.Format("The failed evaluator {0} has been recorded.", evaluatorId);
+                Exceptions.Throw(new IMRUSystemException(msg), Logger);
+            }
+            _failedEvaluators.Add(evaluatorId);
+        }
+
+        /// <summary>
+        /// Checks if the number of failed Evaluators has reached allowed maximum number of evaluator failures 
+        /// </summary>
+        internal bool ReachedMaximumNumberOfEvaluatorFailures
+        {
+            get { return _failedEvaluators.Count >= AllowedNumberOfEvaluatorFailures; }
+        }
+
+        /// <summary>
+        /// Returns allowed maximum number of evaluator failures
+        /// </summary>
+        internal int AllowedNumberOfEvaluatorFailures
+        {
+            get { return _allowedNumberOfEvaluatorFailures; }
+        }
+
+        /// <summary>
+        /// Reset failed Evaluator collection
+        /// </summary>
+        internal void ResetFailedEvalutors()
+        {
+            _failedEvaluators.Clear();
+        }
+
+        /// <summary>
+        /// Sets master Evaluator id.
+        /// Throws IMRUSystemException if evaulatorId is null or _masterEvaluatorId is not null.
+        /// </summary>
+        /// <param name="evaluatorId"></param>
+        private void SetMasterEvaluatorId(string evaluatorId)
+        {
+            if (evaluatorId == null)
+            {
+                Exceptions.Throw(new IMRUSystemException("Master evaluatorId should not be null."), Logger);
+            }
+
+            if (_masterEvaluatorId != null)
+            {
+                string msg = string.Format("There is already a master evaluator {0}", _masterEvaluatorId);
+                Exceptions.Throw(new IMRUSystemException(msg), Logger);
+            }
+            _masterEvaluatorId = evaluatorId;
+        }
+
+        /// <summary>
+        /// Sets master Evaluator id to null.
+        /// If the master Evaluator is already null, throw IMRUSystemException
+        /// </summary>
+        internal void ResetMasterEvaluatorId()
+        {
+            if (_masterEvaluatorId == null)
+            {
+                Exceptions.Throw(new IMRUSystemException("Master evaluator is already null"), Logger);
+            }
+            _masterEvaluatorId = null;
+        }
+
+        /// <summary>
+        /// Checks if the IAllocatedEvaluator is for master
+        /// </summary>
+        /// <param name="evaluator"></param>
+        /// <returns></returns>
+        internal bool IsMasterEvaluator(IAllocatedEvaluator evaluator)
+        {
+            return evaluator.EvaluatorBatchId.Equals(MasterBatchId);
```

Comment:
Renamed to IsEvaluatorForMaster. We need to check if the allocated evaluator is for master.